_episodes_rmd/11-vector-raster-integration.Rmd

```diff
@@ -295,151 +285,126 @@ requires:
-# extract the average tree height (calculated using the raster pixels)
-# located within the AOI polygon
-av_tree_height_AOI <- extract(x = chm_HARV,
+mean_tree_height_AOI <- extract(x = CHM_HARV,
                               y = as(aoi_boundary_HARV, "Spatial"),
-                              fun = mean,
-                              df = TRUE)
-
-# view output
-av_tree_height_AOI
+                              fun = mean)
 
+mean_tree_height_AOI
 ```
 
 It appears that the mean height value, extracted from our LiDAR data derived
-canopy height model is 22.43 meters.
+canopy height model is 19.34 meters.
 
-##Extract Data using x,y Locations
+## Extract Data using x,y Locations
 
 We can also extract pixel values from a raster by defining a buffer or area
 surrounding individual point locations using the `extract()` function. To do this
-we define the summary method (`fun=mean`) and the buffer distance (`buffer=20`)
-which represents the radius of a circular region around each point.
-
-The units of the buffer are the same units of the data `CRS`.
+we define the summary argument (`fun = mean`) and the buffer distance (`buffer = 20`)
+which represents the radius of a circular region around each point. By default, the units of the
+buffer are the same units as the data's CRS.
 
 <figure>
-    <a href="http://neondataskills.org/images/spatialData/BufferCircular.png">
-    <img src="http://neondataskills.org/images/spatialData/BufferCircular.png"></a>
+    <a href="https://www.neonscience.org/sites/default/files/images/spatialData/BufferCircular.png">
+    <img src="https://www.neonscience.org/sites/default/files/images/spatialData/BufferCircular.png"></a>
     <figcaption> Extract raster information using a buffer region. All pixels
     that are touched by the buffer region are included in the extract.
     Source: National Ecological Observatory Network (NEON).
     </figcaption>
 </figure>
 
-Let's put this into practice by figuring out the average tree height in the
-20m around the tower location.
+Let's put this into practice by figuring out the mean tree height in the
+20m around the tower location (`point_HARV`). Because we are extracting only a single number, we
+will not use the `df = TRUE` argument. 
 
 ```{r extract-point-to-buffer }
-# what are the units of our buffer
-st_crs(point_HARV)
-
-# extract the average tree height (height is given by the raster pixel value)
-# at the tower location
-# use a buffer of 20 meters and mean function (fun)
-av_tree_height_tower <- extract(x = chm_HARV,
+mean_tree_height_tower <- extract(x = CHM_HARV,
                                 y = as(point_HARV, "Spatial"),
                                 buffer = 20,
-                                fun = mean,
-                                df = TRUE)
-
-# view data
-head(av_tree_height_tower)
-
-# how many pixels were extracted
-nrow(av_tree_height_tower)
+                                fun = mean)
 
+mean_tree_height_tower
```

Comment:
Is this a good place to also talk about `cellStats()`? It can be used to get fast summaries for whole rasters - mean, median, sd etc like `cellStats(CHM_HARV, mean, na.rm = TRUE)`